### PR TITLE
CON-2508: Analyzing the multipath devices along with volume attachments

### PR DESCRIPTION
**Pull request by AnushaY1916**
In this PR, the node monitor and the node init container will analyze the multipath devices and report whether they belong to this node or not and what further actions need to be done.

### CODE REVIEW

**Review by dileepds — APPROVED** [review]
LGTM

File: pkg/flavor/kubernetes/flavor.go [hunk 1]
```diff
@@ -787,6 +787,8 @@ func (flavor *Flavor) ListVolumeAttachments() (*storage_v1.VolumeAttachmentList,
 	if err != nil {
 		return nil, err
 	}
+	log.Trace("VOLUME ATTACHMENT LIST:", vaList)
+	log.Tracef("VOLUME ATTACHMENT LIST:%+v", vaList)
```
Comment (datamattsson) [inline pkg/flavor/kubernetes/flavor.go:791]: What is the purpose of the first one?

Comment (AnushaY1916) [reply pkg/flavor/kubernetes/flavor.go:791]: These lines were for debugging purposes. Will remove this.

**Review by datamattsson — COMMENTED** [review]
There's a couple of spelling mistakes here and there.

File: pkg/nodeinit/utils.go [hunk 2]
```diff
@@ -2,30 +2,84 @@ package nodeinit
...
+	if err != nil {
+		return err
 	}
+	for _, device := range multipathDevices {
+		if vaList != nil && len(vaList.Items) > 0 {
+			log.Infof("Assessing the multipath device %s", device.Name)
+			if doesDeviceBelongToTheNode(device, vaList, nodeName) {
+				log.Infof("Multipath device %s belongs to the node %s", device.Name, nodeName)
+				if device.IsUnhealthy {
+					log.Info("The multipath device %s belongs to this node %s and is unhealthy. Issue warnings!", device.Name, nodeName)
```
Comment (datamattsson) [inline pkg/nodeinit/utils.go:61]: should this be `log.Infof`?

File: pkg/nodeinit/utils.go [hunk 3]
```diff
@@ -2,30 +2,84 @@ package nodeinit
...
+				} else {
+					log.Info("The multipath device %s belongs to this node %s and is healthy. Nothing to do", device.Name, nodeName)
```
Comment (datamattsson) [inline pkg/nodeinit/utils.go:63]: `log.infof`?

**Review by datamattsson — COMMENTED** [review]
Nits.

**Pull request merged**